core/src/main/java/io/undertow/attribute/RequestSizeAttribute.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2014 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package io.undertow.attribute;
+
+import io.undertow.server.HttpServerExchange;
+
+/**
+ * Size of request in bytes, including headers, cannot be zero
+ *
+ * @author Marek Jusko
+ */
+
+public class RequestSizeAttribute implements ExchangeAttribute{
+
+    public static final String REQUEST_SIZE_SHORT = "%E";
+    public static final String REQUEST_SIZE = "%{REQUEST_SIZE}";
+    public static final ExchangeAttribute INSTANCE = new RequestSizeAttribute();
+
+    @Override
+    public String readAttribute(HttpServerExchange exchange) {
```

Comment:
What if HTTP 2 protocol is in use? The calculations here expect HTTP 1.1 is being used.